proto/proto/tsmt.proto

```diff
@@ -0,0 +1,38 @@
+// Message definitions related to Tiered Sparse Merkle Trees (TSMT).
+//
+// Notes about TSMT:
+// - The tree is a key value store, both are digest values.
+// - The key's most significant byte is used to deterministically assign the
+// pair to one of the tree's entries.
```

Comment:
nit: this should say "most significant element" or "8 most significant bytes".

Also, minor nit on formatting: I'd indent second lines of each bullet point to make them easier to read. For example:
```Rust
// - The tree is a key value store, both are digest values.
// - The key's most significant byte is used to deterministically assign the
//   pair to one of the tree's entries.
```
Instead of:
```Rust
// - The tree is a key value store, both are digest values.
// - The key's most significant byte is used to deterministically assign the
// pair to one of the tree's entries.
```

Reply:
changed